src/select/parts/virtual-list.tsx

```diff
@@ -107,7 +107,10 @@ const VirtualListOpen = forwardRef(
           aria-hidden="true"
           key="total-size"
           className={styles['layout-strut']}
-          style={{ height: totalSize - stickySize }}
+          style={{
+            // Adjust totalSize to account for 1px overlap per item (matching the position adjustment in renderOptions)
+            height: totalSize - filteredOptions.length - stickySize,
```

Comment:
Can we compute the correct totalSize inside the `useVirtual()`? That is a util that we own - we can theoretically parametrise it so that the total size and item sizes area adjusted automatically when we need that 1px overlap feature.